packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -90,41 +109,29 @@ export function createTokenIntrospectionMiddleware({
         throw new OpenPaymentsServerRouteError(403, 'Inactive Token')
       }
 
-      // TODO
-      // https://github.com/interledger/rafiki/issues/835
-      const access = tokenInfo.access.find((access: Access) => {
-        if (
-          access.type !== requestType ||
-          (access.identifier && access.identifier !== ctx.walletAddressUrl)
-        ) {
-          return false
-        }
-        if (
-          requestAction === AccessAction.Read &&
-          access.actions.includes(AccessAction.ReadAll)
-        ) {
-          ctx.accessAction = AccessAction.ReadAll
-          return true
-        }
-        if (
-          requestAction === AccessAction.List &&
-          access.actions.includes(AccessAction.ListAll)
-        ) {
-          ctx.accessAction = AccessAction.ListAll
-          return true
-        }
-        return access.actions.find((tokenAction: AccessAction) => {
-          if (isActiveTokenInfo(tokenInfo) && tokenAction === requestAction) {
-            ctx.accessAction = requestAction
-            return true
-          }
-          return false
-        })
-      })
+      if (tokenInfo.access.length > 1) {
```

Comment:
Whether or not we need this check depends on if it is decided that the token introspection endpoint will always return ONLY the request's matching access item in the response.

Reply:
However, looking at

https://github.com/interledger/rafiki/pull/2788/files#diff-4c4c1504c8978d5fcd7a7bfa8210903c7dada6ec793ac0ad8e895fda2f0a5115R120-R122

since it looks like its "possible" to return multiple accesses, we should keep the original behavior of allowing multiple? Or we make `accessItem` required in the auth introspection route handler.

Reply:
@mkurapov Here's what I decided - the introspection endpoint on the auth server will always return whatever access rights were presented in the request if the token is valid, no more, no less.

The backend service can then expect the response to have exactly one item since the middleware only ever presents one access during the introspection request.

Reply:
@njlie makes sense to me 👍